mono/mini/aot-compiler.c

```diff
@@ -10443,7 +10442,8 @@ static const char *preinited_jit_icalls[] = {
 	"mono_aot_init_gshared_method_vtable",
 	"mono_llvm_throw_corlib_exception",
 	"mono_init_vtable_slot",
-	"mono_helper_ldstr_mscorlib"
+	"mono_helper_ldstr_mscorlib",
+	"mono_jit_thread_attach"
```

Comment:
@vargaz Could you take a second thought at this from what we discussed yesterday. We can avoid the trampolines for an icall by just adding it here (assuming it doesn't require a wrapper, in which case we would resolve the patch target to a jit trampoline). I fail to see why this wouldn't work since it's basically the same thing : on aot the call is made through a got slot that it's initialized at aot-module load time with the address of the native function.

I'm also getting more confused why we would need special handling in the first place. If the icall for these functions doesn't require a wrapper doesn't this mean that the got entry for them will never be a specific/aot trampoline ?